### ISSUE TITLE
setting DigitizedLandmarks/HeadPoints

### ISSUE THREAD

**Issue by monkeyman192** [mne-tools/mne-bids#48]
At the moment both of these values are set as false by default.
Since they are both REQUIRED values I think it would be good if they could be set correctly and automatically.
For KIT data, you can have it so that when you specify something in the `electrode` or `hsp` arguments of `raw_to_bids` you can set the values correctly like so:
```python
    extra_data["DigitizedLandmarks"] = (True if (electrode is not None and emptyroom is not True) else False)
    extra_data["DigitizedHeadPoints"] = (True if (hsp is not None and emptyroom is not True) else False)
```
Where this extra_data dictionary is passed to the requisite function.
Also, see #50  about the empty room variable above.

Both .elp and .hsp files (for KIT data) should also be copied to the working folder (see sec. 8.4.5 of the bids specification)
I currently have working code for KIT data, but this is one thing that will require a bit more work to get working for other formats.

**Comment by jasmainak** [comment]
@teonbrooks what's your opinion?

**Comment by teonbrooks** [comment]
well, this will likely lead to the resurrection of this dead PR (https://github.com/mne-tools/mne-python/pull/1001). I agree that these points should be set automatically if provided with the args. just a matter of getting these files to cooperate.

tl;dr on old pr, could get the device-head transformation within tolerance.

**Comment by monkeyman192** [comment]
I was thinking something that would be nice is if mne actually kept track of the files that were used to produce the raw file. Like for example the `raw.filenames` is a list which (at least for KIT data) is simply just a single .con file. If this contained also the path to the other files that were used to generate the raw object (so in the case of KIT data it would have the path to the .mrk, .elp and .hsp files if they are provided), then this info could be read directly from `raw.filenames`, and all the moving of files could also be handled easily as we would have the paths of all the constituent files too.

**Comment by teonbrooks** [comment]
oh i see. I guess the motivation there was to have a trace of the original data source. the ancillary head shape and marker points just got absorbed in the FIF file with no looking back. I doubt that the filename list would be modified to accommodate other files but I could be wrong. If you want that feature, that would need to be on the mne-python issue page.

**Comment by monkeyman192** [comment]
Yeha, that's fair enough. I was thinking of raising it as an issue. It wouldn't matter so much here for mne-bids if you specify the files directly to `raw_to_bids`, however if you are passing a raw object in instead of a path then you also need to specify the elp and hsp files (and .mrk also) explicitly so all that data can be used to construct the bids data correctly.
Obviously the nicest solution would be to be able to just get out all the data from the raw object, and I guess that would have to implemented in some way as some data types only have that data stored in their raw data and not as separate files like the KIT system.

**Comment by teonbrooks** [comment]
precisely. that is the convenient nature of the FIF file, it maintains all this information in one place. are you ok with closing this issue?

**Comment by monkeyman192** [comment]
But this issue hasn't really been solved? It would still be good to have some way to get mne-bids to determine the correct values for the two variables.

**Comment by teonbrooks** [comment]
my bad, I thought you were just trying to propose the addition of the files to raw.filename. so you're saying that `DigitizedLandmarks` and `DigitizedHeadPoints` aren't being set with KIT data?

**Comment by monkeyman192** [comment]
Those values aren't being set for *any* data:
[See here](https://github.com/mne-tools/mne-bids/blob/master/mne_bids/meg_bids.py#L283-L284)

**Comment by jasmainak** [comment]
okay so PR welcome :)
